tests/torch/nas/creators.py

```diff
@@ -86,6 +86,7 @@ def create_bnas_model_and_ctrl_by_test_desc(desc: MultiElasticityTestDesc):
 
     nncf_config = NNCFConfig.from_dict(config)
     model = desc.model_creator()
+    model.eval()
```

Comment:
@ljaljushkin now the graph building in NNCFNetwork preserves the train/eval state of the original module, while before it forced the model into eval. As a result, the NAS tests were written with implicit reliance on the model being in `.eval()` state. What is worse that the test actually fails in the `.train()` state. I added explicit `.eval()` calls so that the tests are now passing, but you may want to investigate this.

Reply:
I've investigated this issue. The root cause in changing of BN statistics in the train mode, not in the invalid weight reorg.
The test is not incorrect in train mode, since the output of the model is calculated with different statistics before and after weight reorg. 
Reordering indexes and order of updating weights is the same in eval and train modes. Propose to keep the fix with eval mode.